bigtop-packages/src/charm/layer-hive/lib/charms/layer/apache_hive.py

```diff
@@ -0,0 +1,137 @@
+            # props['hive.exec.local.scratchdir'] = "/tmp/hive"
+            # props['hive.downloaded.resources.dir'] = "/tmp/hive_resources"
+            # props['hive.querylog.location'] = "/tmp/hive"
+            # props['hive.server2.logging.operation.log.location'] = "/tmp/hive"
+            props['hive.hwi.war.file'] = "lib/hive-hwi.jar"
+
+    # called during config-changed events
+    def configure_hive(self):
+        config = hookenv.config()
+        hive_env = self.dist_config.path('hive_conf') / 'hive-env.sh'
+        utils.re_edit_in_place(hive_env, {
+            r'.*export HADOOP_HEAPSIZE *=.*': 'export HADOOP_HEAPSIZE=%s' % config['heap'],
+        })
+
+    def configure_remote_db(self, mysql):
+        hive_site = self.dist_config.path('hive_conf') / 'hive-site.xml'
+        jdbc_url = "jdbc:mysql://{}:{}/{}?createDatabaseIfNotExist=true".format(
+            mysql.host(), mysql.port(), mysql.database()
+        )
+        with utils.xmlpropmap_edit_in_place(hive_site) as props:
+            props['javax.jdo.option.ConnectionURL'] = jdbc_url
+            props['javax.jdo.option.ConnectionUserName'] = mysql.user()
+            props['javax.jdo.option.ConnectionPassword'] = mysql.password()
+            props['javax.jdo.option.ConnectionDriverName'] = "com.mysql.jdbc.Driver"
+
+        hive_env = self.dist_config.path('hive_conf') / 'hive-env.sh'
+        utils.re_edit_in_place(hive_env, {
+            r'.*export HIVE_AUX_JARS_PATH *=.*': 'export HIVE_AUX_JARS_PATH=/usr/share/java/mysql-connector-java.jar',
+        })
+
+        # Now that we have db connection info, init our schema (only once)
+        remote_db = hookenv.remote_service_name()
+        if not unitdata.kv().get('hive.schema.initialized.%s' % remote_db):
+            utils.run_as('ubuntu', 'schematool', '-initSchema', '-dbType', 'mysql')
+            unitdata.kv().set('hive.schema.initialized.%s' % remote_db, True)
+
+    def configure_local_db(self):
+        local_url = 'jdbc:derby:;databaseName=/var/lib/hive/metastore/metastore_db;create=true'
+        local_driver = 'org.apache.derby.jdbc.EmbeddedDriver'
+        hive_site = self.dist_config.path('hive_conf') / 'hive-site.xml'
+        with utils.xmlpropmap_edit_in_place(hive_site) as props:
+            props['javax.jdo.option.ConnectionURL'] = local_url
+            props['javax.jdo.option.ConnectionUserName'] = 'APP'
+            props['javax.jdo.option.ConnectionPassword'] = 'mine'
+            props['javax.jdo.option.ConnectionDriverName'] = local_driver
+
+        hive_env = self.dist_config.path('hive_conf') / 'hive-env.sh'
+        utils.re_edit_in_place(hive_env, {
+            r'.*export HIVE_AUX_JARS_PATH *=.*': '# export HIVE_AUX_JARS_PATH=',
+        })
+
+    def restart(self):
+        self.stop()
+        self.start()
+
+    def start(self):
+        self.stop()
+        hive_log = self.dist_config.path('hive_logs') / 'HiveServer2.out'
+        utils.run_bg_as(
```

Comment:
This should be a system service, right? Lets put it as a backlog item to fix it either inside bigtop or using system scripts.